File: DSA-js/queue.js
class Queue{
    constructor(){
        this.items=[];
    }

    enqueue(element){
        this.items.push(element);
    }

    dequeue(){
        if(this.isEmpty()){
            return "Queue is empty";
        }

        return this.items.shift();
    }

    peek(){
        if(this.isEmpty()){
            return "Queue is empty";
        }
        return this.items[0];
    }

    isEmpty(){
        return this.items.length===0;
    }

    size(){
        return this.items.length;
    }
}

const queue = new Queue();
queue.enqueue(10);
queue.enqueue(25);
queue.enqueue(42);
console.log(queue.peek());    
console.log(queue.dequeue()); 
console.log(queue.peek());

File: DSA-js/queue.java
class Queue {
    int[] items;
    int front;
    int rear;

    Queue() {
        items = new int[100];
        front = 0;
        rear = -1;
    }

    void enqueue(int element) {
        rear++;
        items[rear] = element;
    }

    int dequeue() {
        if (isEmpty()) {
            return -1;
        }

        return items[front++];
    }

    int peek() {
        if (isEmpty()) {
            return -1;
        }

        return items[front];
    }

    boolean isEmpty() {
        return front > rear;
    }

    int size() {
        return rear - front + 1;
    }
}

public class Main {
    public static void main(String[] args) {

        Queue queue = new Queue();

        queue.enqueue(10);
        queue.enqueue(25);
        queue.enqueue(42);

        System.out.println(queue.peek());
        System.out.println(queue.dequeue());
        System.out.println(queue.peek());
    }
}